src/random_walk.c

```diff
@@ -196,31 +219,61 @@ int igraph_random_edge_walk(const igraph_t *graph,
                 IGRAPH_ERROR("Random walk got stuck", IGRAPH_ERWSTUCK);
         }
 
-        if (weights) { /* weighted: choose an out-edge with probability proportional to its weight */
+        if (weights) /* weighted: choose an out-edge with probability proportional to its weight */
+        {
             igraph_real_t r;
-            long j;
+            igraph_vector_t **cd = (igraph_vector_t **) &(VECTOR(cdfs)[start]);
 
-            IGRAPH_CHECK(igraph_vector_resize(&ws, degree));
-            for (j=0; j < degree; ++j)
-                VECTOR(ws)[j] = VECTOR(*weights)[ VECTOR(*edges)[j] ];
+            /* compute out-edge cdf for this node if not already done */
+            if (IGRAPH_UNLIKELY(! *cd)) {
+                long j;
 
-            IGRAPH_CHECK(igraph_vector_cumsum(&cs, &ws));
+                *cd = igraph_malloc(sizeof(igraph_vector_t));
```

Comment:
You need to check whether `igraph_malloc` returned null and return with `IGRAPH_ENOMEM` if so.

Reply:
Sorry about the slow response.  Would this look reasonable (message phrasing, etc.)?

                if (*cd == NULL)
                    IGRAPH_ERROR("random edge walk failed", IGRAPH_ENOMEM);

Reply:
Sure, that's fine.